client/my-sites/plugins/controller.js

```diff
@@ -147,3 +147,28 @@ export function scrollTopIfNoHash( context, next ) {
 	}
 	next();
 }
+
+export function redirectOnUnknownCategory( context, next ) {
+	const category = context.params.category;
+	const site = context.params.site;
+
+	if ( ! ALLOWED_CATEGORIES.includes( category ) ) {
+		const redirectionUrl = `/plugins/${ site || '' }`;
+		page.redirect( redirectionUrl );
+	}
+
+	next();
+}
+
+export function redirectOnSearchInDiscoveryCategories( context, next ) {
+	const category = context.params.category;
+	const searchTerm = context.query.s;
+	const site = context.params.site;
+
+	if ( searchTerm && DISCOVERY_CATEGORIES.includes( category ) ) {
+		const redirectionUrl = `/plugins/${ site || '' }?s=${ searchTerm }`;
+		page.redirect( redirectionUrl );
+	}
```

Comment:
Do we need this redirect anymore if we're doing this: p1661381701763909/1661372850.310569-slack-CKZHG0QCR ?

Reply:
You're right, I'll remove it!